### PR TITLE
chore: reset page/context for reuse in component tests

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/playwright-core/src/server/chromium/crPage.ts [hunk 1]
```diff
@@ -798,10 +808,19 @@ class FrameSession {
...
     ]);
+    this._exposedBindingNames.push(binding.name);
+    this._evaluateOnNewDocumentIdentifiers.push(response.identifier);
+  }
+
+  async _removeExposedBindings() {
+    const names = this._exposedBindingNames;
+    this._exposedBindingNames = [];
+    for (const name of names)
+      await this._client.send('Runtime.removeBinding', { name });
```
Comment (dgozman) [inline packages/playwright-core/src/server/chromium/crPage.ts:823]: nit: might as well `Promise.all`

File: packages/playwright-core/src/server/chromium/crPage.ts [hunk 2]
```diff
@@ -1054,7 +1073,15 @@ class FrameSession {
...
-    await this._client.send('Page.addScriptToEvaluateOnNewDocument', { source, worldName });
+    const { identifier } = await this._client.send('Page.addScriptToEvaluateOnNewDocument', { source, worldName });
+    this._evaluateOnNewDocumentIdentifiers.push(identifier);
+  }
+
+  async _removeEvaluatesOnNewDocument(): Promise<void> {
+    const identifiers = this._evaluateOnNewDocumentIdentifiers;
+    this._evaluateOnNewDocumentIdentifiers = [];
+    for (const identifier of identifiers)
+      await this._client.send('Page.removeScriptToEvaluateOnNewDocument', { identifier });
```
Comment (dgozman) [inline packages/playwright-core/src/server/chromium/crPage.ts:1084]: ditto

File: packages/playwright-core/src/server/page.ts [hunk 3]
```diff
@@ -416,6 +423,10 @@ export class Page extends SdkObject {
     await this._delegate.addInitScript(source);
   }
 
+  async removeInitScripts() {
+    await this._delegate.removeInitScripts();
```
Comment (dgozman) [inline packages/playwright-core/src/server/page.ts:427]: Do we miss `this.initScripts = []` here?

File: packages/playwright-core/src/server/webkit/wkPage.ts [hunk 4]
```diff
@@ -762,6 +766,10 @@ export class WKPage implements PageDelegate {
     await this._updateBootstrapScript();
   }
 
+  async removeInitScripts() {
```
Comment (dgozman) [inline packages/playwright-core/src/server/webkit/wkPage.ts:769]: I struggle to see who calls this one 😄

File: packages/playwright-test/src/mount.ts [hunk 5]
```diff
@@ -14,10 +14,16 @@
...
 
-import type { Page } from '@playwright/test';
+import type { Page, ViewportSize } from '@playwright/test';
 import { createGuid } from 'playwright-core/lib/utils/utils';
 
-export async function mount(page: Page, jsxOrType: any, options: any): Promise<string> {
+export async function mount(page: Page, jsxOrType: any, options: any, baseURL: string, viewport: ViewportSize): Promise<string> {
+  await page.goto('about:blank');
+  await (page as any)._resetForReuse();
+  await (page.context() as any)._resetForReuse();
```
Comment (dgozman) [inline packages/playwright-test/src/mount.ts:23]: Should we close all other pages here?

**Review by dgozman — APPROVED** [review]

File: packages/playwright-core/src/server/page.ts (hunk 3 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/page.ts:427]: Ouch

File: packages/playwright-core/src/server/webkit/wkPage.ts (hunk 4 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/webkit/wkPage.ts:769]: await this._delegate.removeInitScripts() in Page;

File: packages/playwright-test/src/mount.ts (hunk 5 above)
Comment (pavelfeldman) [reply packages/playwright-test/src/mount.ts:23]: I was debating it, dunno.

**Pull request merged**